In [1]:
# data analysis

import pandas as pd
import numpy as np
import random

# Visualization

import seaborn as sb
import matplotlib
import matplotlib.pyplot
%matplotlib inline



In [2]:
train_df = pd.read_csv("train.csv")
test_df = pd.read_csv("test.csv")

In [3]:
train_df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [4]:
train_df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [5]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


In [6]:
train_df.describe( include="all")

PassengerId    Survived      Pclass                             Name  \
count    891.000000  891.000000  891.000000                              891   
unique          NaN         NaN         NaN                              891   
top             NaN         NaN         NaN  Hagland, Mr. Ingvald Olai Olsen   
freq            NaN         NaN         NaN                                1   
mean     446.000000    0.383838    2.308642                              NaN   
std      257.353842    0.486592    0.836071                              NaN   
min        1.000000    0.000000    1.000000                              NaN   
25%      223.500000    0.000000    2.000000                              NaN   
50%      446.000000    0.000000    3.000000                              NaN   
75%      668.500000    1.000000    3.000000                              NaN   
max      891.000000    1.000000    3.000000                              NaN   

         Sex         Age       SibSp       Parch Ticket        Fare  \
count    891  714.000000  891.000000  891.000000    891  891.000000   
unique     2         NaN         NaN         NaN    681         NaN   
top     male         NaN         NaN         NaN   1601         NaN   
freq     577         NaN         NaN         NaN      7         NaN   
mean     NaN   29.699118    0.523008    0.381594    NaN   32.204208   
std      NaN   14.526497    1.102743    0.806057    NaN   49.693429   
min      NaN    0.420000    0.000000    0.000000    NaN    0.000000   
25%      NaN   20.125000    0.000000    0.000000    NaN    7.910400   
50%      NaN   28.000000    0.000000    0.000000    NaN   14.454200   
75%      NaN   38.000000    1.000000    0.000000    NaN   31.000000   
max      NaN   80.000000    8.000000    6.000000    NaN  512.329200   

              Cabin Embarked  
count           204      889  
unique          147        3  
top     C23 C25 C27        S  
freq              4      644  
mean            NaN      NaN  
std             NaN      NaN  
min             NaN      NaN  
25%             NaN      NaN  
50%             NaN      NaN  
75%             NaN      NaN  
max             NaN      NaN

In [7]:
train_df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [8]:
train_df.describe(include = 'O')

Name   Sex Ticket        Cabin Embarked
count                               891   891    891          204      889
unique                              891     2    681          147        3
top     Hagland, Mr. Ingvald Olai Olsen  male   1601  C23 C25 C27        S
freq                                  1   577      7            4      644

Temos PassengerId com 891 registros distintos
As colunas Survived , Pclass , Name , Sex , SibSp Parch , Ticket , Fare não possuem valores NULL
As colunas Age , Cabin , Embarked possuem valores Null
A coluna Ticket pode ser excluida

In [9]:
i = 0

In [10]:
train_df['Survived'].count()

891

In [11]:
while i < train_df['Survived'].count():
    if train_df['Age'][i] > 0 and train_df['Age'][i] < 11:
        train_df['Age_Group'] = 0
    if train_df['Age'][i] > 10 and train_df['Age'][i] < 21:
        train_df['Age_Group'] = 1
    if train_df['Age'][i] > 20 and train_df['Age'][i] < 31:
        train_df['Age_Group'] = 2         
    if train_df['Age'][i] > 30 and train_df['Age'][i] < 41:
        train_df['Age_Group'] = 3
    if train_df['Age'][i] > 40 and train_df['Age'][i] < 51:
        train_df['Age_Group'] = 4
    if train_df['Age'][i] > 50 and train_df['Age'][i] < 61:
        train_df['Age_Group'] = 5
    if train_df['Age'][i] > 60 and train_df['Age'][i] < 71:
        train_df['Age_Group'] = 6
    if train_df['Age'][i] > 70 and train_df['Age'][i] < 81:
        train_df['Age_Group'] = 1
    if train_df['Age'][i] > 80 and train_df['Age'][i] < 91:
        train_df['Age_Group'] = 1
    #print( train_df['Age'][i])
    i = i + 1

In [12]:
features = train_df.columns.tolist()

In [10]:
print(features)

['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


In [11]:
features.pop(features.index('PassengerId'))
features.pop(features.index('Name'))
features.pop(features.index('Ticket'))
features.pop(features.index('Cabin'))

'Cabin'

In [12]:
print(features)

['Survived', 'Pclass', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Embarked']


In [17]:
#train_df['Survived'].count()
train_df.count()

PassengerId    891
Survived       891
Pclass         891
Name           891
Sex            891
Age            714
SibSp          891
Parch          891
Ticket         891
Fare           891
Cabin          204
Embarked       889
dtype: int64

In [ ]:
train_df['Survived'].count()

In [42]:
train_df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  Age_Group  
0      0         A/5 21171   7.2500   NaN        S          3  
1      0          PC 17599  71.2833   C85        C          3  
2      0  STON/O2. 3101282   7.9250   NaN        S          3  
3      0            113803  53.1000  C123        S          3  
4      0            373450   8.0500   NaN        S          3

In [34]:
combine = [train_df, test_df]

In [ ]:
X = pd.concat( [train_df[['Pclass' , 'Age' , 'SibSp' , 'Parch' , 'Fare' ]] ,  pd.get_dummies(train_df[[ 'Sex' , 'Embarked' ]])], axis=1 )

In [ ]:
X.head()